windows/sound.cpp

```diff
@@ -80,15 +87,26 @@ QString CSound::LoadAndInitializeDriver ( QString strDriverName,
     // check if device is capable
     if ( strStat.isEmpty() )
     {
-// TODO: In order to fix https://github.com/jamulussoftware/jamulus/issues/796 we reset the channel mapping on every property change. This is not ideal. 
-	    
-// the device has changed, per definition we reset the channel
-// mapping to the defaults (first two available channels)
-ResetChannelMapping();
-
-// store ID of selected driver if initialization was successful
-strCurDevName = cDriverNames[iDriverIdx];
-
+        // Reset channel mapping if the number of channels has changed or the sound card name has changed
+        if ( ( lNumInChanPrev != lNumInChan ) ||

```

Comment:
I'd make the logic clearer:
- If _(_ the driver name changed _)_ _OR_ _(_ the indexes changed _)_

(and swap the comment around -- soundcard changing takes precedence over inputs).